src/Mvc/Mvc.Core/src/Properties/AssemblyInfo.cs

```diff
@@ -3,8 +3,11 @@
 
 using System.Runtime.CompilerServices;
 using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.AspNetCore.Mvc.Routing;
 
 [assembly: TypeForwardedTo(typeof(InputFormatterException))]
+[assembly: TypeForwardedTo(typeof(IRouteTemplateProvider))]
+[assembly: TypeForwardedTo(typeof(IActionHttpMethodProvider))]
```

Comment:
We already have this https://github.com/dotnet/aspnetcore/blob/main/src/Http/Routing/src/IHttpMethodMetadata.cs, why do we need `IActionHttpMethodProvider`?

Reply:
I always wondered why the attributes in MVC didn't implement this directly. @rynowak why did we need the indirection?

Reply:
I was just looking at what was already implemented on `[HttpGet]`, `[HttpPost]` etc...

I can make those also implement IHttpMethodMetadata so we don't need to use or type forward IActionHttpMethodProvider. Would you have the attribute implementations of IHttpMethodMetadata always return false from the IHttpMethodMetadata.AcceptCorsPreflight?

Reply:
> MVC didn't implement this directly

MVC already had these APIs before endpoint routing was created.

Reply:
> Would you have the attribute implementations of IHttpMethodMetadata always return false from the IHttpMethodMetadata.AcceptCorsPreflight?

Yes, that's what MVC does internally: https://github.com/dotnet/aspnetcore/blob/c925f99cddac0df90ed0bc4a07ecda6b054a0b02/src/Mvc/Mvc.Core/src/ApplicationModels/DefaultApplicationModelProvider.cs#L657-L673

Reply:
Cool. That's where I copied the logic from, but I just wanted to make sure we didn't want to add a property to the attribute or something.

Reply:
We could have added it in 3.x

Reply:
I think I wanted to avoid implementing a contract without really implementing it. You know, YA BOY LISKOV - the one good SOLID principle. 

MVC's attributes have no opinion about CORs.

It was only after we built MVC that I understood that CORs and HTTP methods are so tightly coupled, hence the different contract.